shell/platform/darwin/macos/framework/Source/FlutterMacOSGLCompositor.mm

```diff
@@ -0,0 +1,94 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterMacOSGLCompositor.h"
+
+#import <OpenGL/gl.h>
+#import "flutter/fml/logging.h"
+#import "flutter/fml/platform/darwin/cf_utils.h"
+#import "third_party/skia/include/core/SkCanvas.h"
+#import "third_party/skia/include/core/SkSurface.h"
+#import "third_party/skia/include/gpu/gl/GrGLAssembleInterface.h"
+#import "third_party/skia/include/utils/mac/SkCGUtils.h"
+
+#include <unistd.h>
+
+namespace flutter {
+
+FlutterMacOSGLCompositor::FlutterMacOSGLCompositor(FlutterViewController* view_controller,
+                                                   NSOpenGLContext* open_gl_context)
+    : view_controller_(view_controller), open_gl_context_(open_gl_context) {}
+
+FlutterMacOSGLCompositor::~FlutterMacOSGLCompositor() = default;
+
+bool FlutterMacOSGLCompositor::CreateBackingStore(const FlutterBackingStoreConfig* config,
+                                                  FlutterBackingStore* backing_store_out) {
+  return CreateBackingStoreUsingSurfaceManager(config, backing_store_out);
+}
+
+bool FlutterMacOSGLCompositor::CollectBackingStore(const FlutterBackingStore* backing_store) {
+  // The memory for FlutterSurfaceManager is handled in the destruction callback.
+  // No other memory has to be collected.
+  return true;
+}
+
+bool FlutterMacOSGLCompositor::Present(const FlutterLayer** layers, size_t layers_count) {
+  for (size_t i = 0; i < layers_count; ++i) {
+    const auto* layer = layers[i];
+    FlutterBackingStore* backing_store = const_cast<FlutterBackingStore*>(layer->backing_store);
+    switch (layer->type) {
+      case kFlutterLayerContentTypeBackingStore: {
+        FlutterSurfaceManager* surfaceManager =
+            (__bridge FlutterSurfaceManager*)backing_store->user_data;
+
+        CGSize size = CGSizeMake(layer->size.width, layer->size.height);
+        [view_controller_.flutterView frameBufferIDForSize:size];
+        [surfaceManager setLayerContentWithIOSurface:[surfaceManager getIOSurface]];
+        break;
+      }
+      case kFlutterLayerContentTypePlatformView:
+        // Add functionality in follow up PR.
+        FML_CHECK(false) << "Presenting PlatformViews not yet supported";
+        break;
+    };
+  }
+  return present_callback_();
+}
+
+bool FlutterMacOSGLCompositor::CreateBackingStoreUsingSurfaceManager(
+    const FlutterBackingStoreConfig* config,
+    FlutterBackingStore* backing_store_out) {
+  FlutterSurfaceManager* surfaceManager =
+      [[FlutterSurfaceManager alloc] initWithLayer:view_controller_.flutterView.layer
+                                     openGLContext:open_gl_context_];
+
+  GLuint fbo = [surfaceManager getFramebuffer];
```

Comment:
getFrameBuffer -> getFrontBuffer

Reply:
Removed IOSurface getter, not sure if this is still necessary for just fbo/texture.

Reply:
Refactored FlutterSurfaceManager, this is not applicable anymore.